src/storage/v2/durability/metadata.hpp

```diff
@@ -48,6 +49,7 @@ struct RecoveredIndicesAndConstraints {
     std::vector<EdgeTypeId> edge;
     std::vector<std::pair<EdgeTypeId, PropertyId>> edge_property;
     std::vector<std::pair<std::string, LabelId>> text_indices;
+    std::vector<std::shared_ptr<VectorIndexSpec>> vector_indices;
```

Comment:
I think its fine that this is a copy of the Spec, not a shared_ptr to the same common entry that was inserted.